proto/groups/v1alpha/groups.proto

```diff
@@ -0,0 +1,204 @@
+// Read the comments!
+message UpdateGroupRequest {
+  // ID of the group to update.
+  string id = 1;
+
+  // Required. Title of the Group.
+  // Must always be provided, even if unchanged.
+  string title = 2;
+
+  // Required. Description of the Group. Can be empty string.
+  // Must always be provided, even if unchanged.
+  string description = 3;
+
+  // Optional. List of members to be updated in the Group.
+  // Key is the member's Account ID,
+  // value is the Role.
+  // To remove a member from the group, set the role to unspecified.
+  map<string, Role> updated_members = 4;
+
+  // Optional. List of content to be updated in the Group.
+  // Key is a pretty path on which the content is published,
+  // value is a Hyperdocs URL of the content.
+  // To unpublish content set the pretty path to an empty string.
+  map<string, string> updated_content = 5;
+}
+
+// Response to list members.
+message ListMembersRequest {
+  // Required. ID of the group to list members of.
+  string id = 1;
+
+  // Optional. Version of the group to list members of.
+  // If not specified, the latest version of the group
+  // according to its members is used.
+  string version = 2;
+
+  // Optional. Maximum number of members to return.
+  int32 page_size = 3;
+
+  // Optional. Page token to continue listing members from.
+  string page_token = 4;
+}
+
+// Response to list members.
+message ListMembersResponse {
+  // ID of the group owner.
+  string owner_account_id = 1;
+
+  // List of members of the group.
+  map<string, Role> members = 2;
+
+  // Token to continue listing members from.
+  string next_page_token = 3;
+}
+
+// Request to list content.
+message ListContentRequest {
+  // Required. ID of the group to list content of.
+  string id = 1;
+
+  // Optional. Version of the group to list content of.
+  // If not specified, the latest version of the group
+  // according to its members is used.
+  string version = 2;
+
+  // Optional. Maximum number of content to return.
+  int32 page_size = 3;
+
+  // Optional. Page token to continue listing content from.
+  string page_token = 4;
+}
+
+// Response to list content.
+message ListContentResponse {
+  // List of content of the group.
+  // Key is the path and value is a Hyperdocs URL to the content.
+  map<string, string> content = 1;
+
+  // Token to continue listing content from.
+  string next_page_token = 2;
+}
+
+// Request to list groups.
+message ListGroupsRequest {
+  // Optional. Maximum number of groups to return.
+  int32 page_size = 1;
+
+  // Optional. Page token to continue listing groups from.
+  string page_token = 2;
+}
+
+// Response to list groups.
+message ListGroupsResponse {
+  // List of groups.
+  repeated Group groups = 1;
+
+  // Token to continue listing groups from.
+  string next_page_token = 2;
+}
+
+// Group represents the basic information about a group entity.
+message Group {
+  // ID of the group.
+  string id = 1;
+
+  // Title of the group.
+  string title = 3;
```

Comment:
where is item no#2? should not be `string title = 2;`?

Reply:
It's ok to have gaps with field numbers. They don't even need to be sequential or anything. Just unique. These numbers are like field names in a JSON object, so it can be any number at all.

Lower numbers are used of course because they use less space in binary encoding.

So, I probably had some field with number 2, but then decided it wasn't necessary.